File: BackEnd/FinalProject/FinalProject.Clean/Models/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinalProject.Core.Models
{
    public class Message
    {
        [Key]
        public int MessageId { get; set; }
        [ForeignKey("User")]
        public int UserId { get; set; }
        public int SenderId { get; set; }
        [MaxLength(250)]
        public string MessageContent { get; set; }
        public DateTime MessageDate { get; set; }

        public Message(int messageId, int userId, int senderId, string messageContent, DateTime messageDate)
        {
            MessageId = messageId;
            UserId = userId;
            SenderId = senderId;
            MessageContent = messageContent;
            MessageDate = messageDate;
        }

        public Message(int userId, int senderId, string messageContent, DateTime messageDate)
        {
            UserId = userId;
            SenderId = senderId;
            MessageContent = messageContent;
            MessageDate = messageDate;
        }
    }
}

File: FrontEnd/ReactUser/src/models/Message.ts
export class Message {
    MessageId: number;
    UserId: number;
    SenderId: number;
    MessageContent: string;
    MessageDate: Date;
    constructor(data: Partial<Message> = {}) {
        this.MessageId = data.MessageId ?? 0;
        this.UserId = data.UserId ?? 0;
        this.SenderId = data.SenderId ?? 0;
        this.MessageContent = data.MessageContent ?? "defalt content";
        this.MessageDate = data.MessageDate ?? new Date();
    }
}

export class MessagePost {
    UserId: number;
    SenderId: number;
    MessageContent: string;
    MessageDate: Date;
    constructor(data: Partial<Message> = {}) {
        this.UserId = data.UserId ?? 0;
        this.SenderId = data.SenderId ?? 0;
        this.MessageContent = data.MessageContent ?? "defalt content";
        this.MessageDate = data.MessageDate ?? new Date();
    }
}
